src/Tokenizers/PHP.php

```diff
@@ -553,6 +553,23 @@ protected function tokenize($string)
                 $lastNotEmptyToken = ($newStackPtr - 1);
             }
 
+            /*
+                For Explicit Octal Notation prior to PHP 8.1 we need to combine the
+                T_NUMBER and T_STRING token values into a single token value, and
+                then ignore the T_STRING token.
+            */
+
+            if (PHP_VERSION_ID < 80100
+                && $tokenIsArray === true && $token[1] === '0'
+                && (isset($tokens[($stackPtr + 1)]) === true
+                && is_array($tokens[($stackPtr + 1)]) === true
+                && $tokens[($stackPtr + 1)][0] === T_STRING
+                && $tokens[($stackPtr + 1)][1][0] === 'o')
```

Comment:
This does not account for uppercase `O`, which I'm pretty sure is valid (it is for the other explicit annotations).